src/contexts/RestaurantsContext.tsx

```diff
@@ -125,25 +130,28 @@ export const RestaurantsProvider: FC<PropsWithChildren> = ({ children }) => {
         setSelectedVenueTypes(selectedVenueTypes.filter((type: VenueType) => type.id !== venueType.id));
     };
     const restaurantsFiltered: Restaurant[] = selectedOptions.length === 0 && selectedVenueTypes.length === 0 ? restaurantsOnMap : restaurantsOnMap.filter((restaurant) => restaurant.meals.some((meal) => optionNames.includes(meal.name.toLowerCase())) || optionNames.includes(restaurant.name.toLowerCase()) || typeNames.includes(restaurant.type.toLowerCase()));
+
+    const { isPending, isError, isSuccess, data } = useQuery({
+        queryKey: ['restaurants'],
+        queryFn: () => restaurantsService.getRestaurants(),
+    });
+
     useEffect(() => {
-        const fetchRestaurants = async () => {
-            const res = await restaurantsService.getRestaurants();
-            if (res.status === 'error') {
-                setRestaurantsOnMap([]);
-                setIsLoading(false);
-            } else {
-                setRestaurantsOnMap(res.data);
-                setIsLoading(false);
-            }
-        };
-        fetchRestaurants();
-    }, []);
+        if (isError) {
+            setRestaurantsOnMap([]);
+        }
+        if (isSuccess && 'data' in data) {
```

Comment:
```suggestion
        } else if (isSuccess && 'data' in data) {
```
Когда else-if, тогда понятно, что ты считаешь что только одна ветка выполнится

Reply:
спасибо за коммент, но useEffect я полностью удалил